### ISSUE TITLE
API: to_excel with merge_cells=False treats index and columns differently

### ISSUE THREAD

**Issue by rhshadrach** [pandas-dev/pandas#60274]
```python
df = pd.DataFrame({"a": [1, 1], "b": [2, 3], "c": 4, "d": 5}).set_index(["a", "b"])
df.columns = pd.MultiIndex.from_tuples([("x", "y"), ("x", "z")])
df.to_excel("test.xlsx", merge_cells=False)
```

![image](https://github.com/user-attachments/assets/7d766e73-6869-4af5-9abe-e66e1f9dcaf6)

In the above the multi-level nature of the index is maintained, however the columns are collapsed to a single row in the form `x.y, x.z`. It seems to me that these should be treated the same. The output should be multiple rows of non-merged cells.

**Comment by rhshadrach** [comment]
@WillAyd @mroeschke - any thoughts?

**Comment by WillAyd** [comment]
I think it's ok and maybe doesn't even need a deprecation (?)

**Comment by mroeschke** [comment]
Yeah I would be OK to treat this as a "bug fix" especially for 3.0
